core/controllers/profile.py

```diff
@@ -361,6 +364,7 @@ def post(self):
 class SiteLanguageHandler(base.BaseHandler):
     """Changes the preferred system language in the user's preferences."""
 
+    @acl_decorators.open_access
```

Comment:
Yes, that sgtm. Make sure to test manually in both logged-out and logged-in mode to ensure there are no unexpected errors.